src/components/Select.js

```diff
@@ -56,14 +58,18 @@ class Select extends React.Component {
 
     this.state = {
       isOpen: false,
-      selected: props.selected
+      selected: props.selected,
+      options: props.options,
     };
   }
 
   componentWillReceiveProps (newProps) {
     if (!_isEqual(newProps.selected, this.props.selected)) {
       this.setState({ selected: newProps.selected });
     }
+    if(!_isEqual(newProps.options, this.props.options)) {
+      this.setState({ options: newProps.options })
+    }
```

Comment:
We're not big fans of handing props off to state. Is there a specific reason you had for doing this? I would think you would be able to just filter the prop directly for mapping over it to render them rather than doing this.

Reply:
I think in this case it is probably ok. There needs to be a key change event here, and it either gets handled in state here or by component state or redux somewhere else

Reply:
Thanks for pointing this out.
I thought this would be the most simple solution for updating options.

I was (still am) struggling to determine how to update the props for mapping after the `searchInput` callback is fired and props are filtered.  Without triggering by setting state.

Reply:
Actually on further thought I think this should be handled differently.
The props coming in should initialize state. From there on out state should handle the options. 
It will be super weird if the person is typing to filter, and new props come in.

Reply:
Why can't it be something like this?

```javascript
onSearchInputChange = e => {
  this.setState({ searchTerm: e.target.value })
}

render(){
  return (
    <searchInput onChange={this.onSearchInputChange} />
    {
      this.props.options
        .filter(option => {
          return searchTerm ? option.displayValue.toLowerCase().includes(searchTerm) : true
        })
        .map(option => JSX)
    }
  )
}

Reply:
@paniclater 
If I am not mistaken..
By declaring `options: props.options,` in the constructor, we are letting state handle options.

In `componentWillReceiveProps` we should be comparing this.state.options.
And we should refactor the use of `this.props.options` for `this.state.options`.

Reply:
@cerinman 
As you suggested that was another way of handling this.  Adding`searchTerm` as state in this component. 
But It felt that this component should not be responsible for keeping track of `searchTerm` as state, and it should be handled within `searchInput`. 
🤷‍♂

Reply:
I would prefer the suggested method over handing props off to state. Equality checks in life cycle methods tend to be more costly than just running a filter on every render. Equality checks also have a tendency to introduce bugs because of their complexity and/or just not realizing something isn't a deep equality check.

Examples being objects within objects that have arrays of more objects or a check just checking the objects id rather than checking actual key/value pairs within an object. That might not be the case currently but this type of implementation opens the door for such things if some one passes an option object that isn't laid out the way you would expect. 

And even if you ensure deep equality checking, you usually do so at the cost of computational time. Also equality checks on props/state can also cause additional renders that you do not expect. The component may only use one or two key/values pairs  but the option object may be more complex. Changing of an unused key/value pair will still cause the component to re render and that can cause confusion when trying to debug.

All this adds up to a larger surface area to cause problems. The filter method suggested is not a guarantee against any of this of course but it tends to be way less risky.